runtime/compiler/control/CompilationThread.cpp

```diff
@@ -6824,11 +6826,24 @@ TR::CompilationInfoPerThreadBase::preCompilationTasks(J9VMThread * vmThread,
       {
       _vm = TR_J9VMBase::get(_jitConfig, vmThread, TR_J9VMBase::AOT_VM);
       entry->_useAotCompilation = true;
+      // if in JITaaS mode, it's going to be a remote AOT compilation request
+      if (_compInfo.getPersistentInfo()->getJITaaSMode() == CLIENT_MODE)
+         {
+         entry->setRemoteCompReq();
+         }
       }
    else if (entry->isOutOfProcessCompReq())
       {
-      _vm = TR_J9VMBase::get(_jitConfig, vmThread, TR_J9VMBase::J9_SERVER_VM);
-      entry->_useAotCompilation = false;
+      // is this really needed? TR::CompilationInfoPerThreadRemote::processEntry(...) already sets the _vm for us
+      // feels like this is redundant code for JITaaS Server, in my opinion we should remove this one, as the other one might be necessary
+      if (entry->_useAotCompilation)
```

Comment:
I agree that we can avoid setting  the VM again. Needs serious testing though in all variations (with AOT/without, etc)
We should keep a comment that the VM has been set in processEntry()